Reference/conformance/conformance.pb.swift

```diff
@@ -265,7 +265,7 @@ struct Conformance_ConformanceRequest: SwiftProtobuf.Message, SwiftProtobuf.Prot
       self = .None
     }
 
-    public mutating func decodeField(setter: inout SwiftProtobuf.FieldDecoder, protoFieldNumber: Int) throws {
+    public mutating func decodeField<T: SwiftProtobuf.FieldDecoder>(setter: inout T, protoFieldNumber: Int) throws {
```

Comment:
It's a performance boost because it allows you to avoid creating objects elsewhere.

With the old pattern, you had to create a `FieldDecoder` protocol witness to pass into `decodeField`, which led to patterns like this:
```
      // In decoder
      var fieldDecoder: FieldDecoder = self
      message.decodeField(setter: &fieldDecoder, ...)
```
(or worse, creating a new object for the field decoder).  This both copies the decoder object, and also creates an additional protocol witness.  With the generic form, the decoder can just implement the `FieldDecoder` protocol and be passed directly in without copying and without creating a witness.

By my measurement, this adds only a few bytes per field to the generated compiled code.  (17 bytes per field in the one measurement I clearly remember)  I've not attempted to look carefully at the runtime library size, but I don't expect much impact there either.

If you want to dig more carefully into the implications, try checking out 5ce85d4 (just before I made the switch to the generic form) and 	c0f1a33 (just after).